charts/community-operator/templates/operator.yaml

```diff
@@ -70,17 +59,32 @@ spec:
           image: {{ .Values.registry.operator }}/{{ .Values.operator.operatorImageName }}:{{ .Values.operator.version }}
           imagePullPolicy: {{ .Values.registry.pullPolicy}}
           name: {{ .Values.operator.deploymentName }}
-          resources:
-            {{- toYaml .Values.operator.resources | nindent 12 }}
-          {{- if .Values.operator.securityContext }}
-          securityContext:
-            {{- toYaml .Values.operator.securityContext | nindent 12 }}
+          {{- with .Values.operator.resources }}
+          resources: {{- toYaml . | nindent 12 }}
           {{- end }}
-      {{- if .Values.operator.priorityClassName }}
-      priorityClassName: {{ .Values.operator.priorityClassName }}
+          {{- with .Values.operator.securityContext }}
+          securityContext: {{- toYaml . | nindent 12 }}
+          {{- end }}
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.operator.nodeSelector }}
+      nodeSelector: {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.operator.priorityClassName }}
+      priorityClassName: {{ . | quote }}
       {{- end }}
-      {{- if .Values.operator.podSecurityContext }}
-      securityContext:
-        {{- toYaml .Values.operator.podSecurityContext | nindent 8 }}
+      {{- with .Values.operator.schedulerName }}
+      schedulerName: {{ . | quote }}
+      {{- end }}
+      {{- with .Values.operator.podSecurityContext }}
+      securityContext: {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ .Values.operator.name }}
+      {{- with .Values.operator.tolerations }}
```

Comment:
This seems to be implementing tolerations for the operator, which is only a subset of the chart, right? I tried to add tolerations myself and I'm not sure how to add tolerations to the custom resource definition

Reply:
Not sure to understand: this is to schedule the operator pod itself and not the MongoDB pods with CRDs?

Reply:
ah, I thought the intention was to add tolerations also to the mongoDB pods.

I've found a problem when deploying mongoDB to an ARM based environment in GCP. GCP taints [their new ARM based nodes](https://cloud.google.com/compute/docs/general-purpose-machines#t2a_machines) with a specific arm64 tag/label, and the only way I found to deploy pods into them was adding this toleration:

```
  tolerations:
    - key: kubernetes.io/arch
      operator: Equal
      value: arm64
      effect: NoSchedule
```

Unfortunately, I haven't been able to deploy mongodb to an GCP arm64 environment because of this problem. I forked this repo and added the ability to set tolerations for the operator, and that didn't fix my issue. I tried to figure how to also add tolerations to CRDs but I didn't know how.

Should I open a different issue? Am I doing something very wrong here?

Reply:
I will try to make some time to validate the helm chart is fully functional, I didn't try on different architectures but should not be a problem.